api/src/main/java/org/pmiops/workbench/workspaces/WorkspaceServiceImpl.java

```diff
@@ -526,4 +529,13 @@ public UserRecentWorkspace updateRecentWorkspaces(long workspaceId, long userId,
       return recentWorkspace;
     }
   }
+
+  @Override
+  public void maybeDeleteRecentWorkspace(long workspaceId, long userId) {
+    List<UserRecentWorkspace> maybeRecentWorkspaces = userRecentWorkspaceDao.findByWorkspaceIdAndUserId(workspaceId, userId);
```

Comment:
I don't understand why there'd ever be more than one such entry. I.e. I'd expect to see `Optional<UserRecentWorkspace>` for this function. (Maybe you need to limit 1 on the query for that to work).